src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -22,19 +35,24 @@
           type="text"
           className="input"
           placeholder="Search..."
+          value={filter.query}
+          onChange={e => dispatch(actions.setSearchQuery(e.target.value))}
```

Comment:
Ensure that `filter.query` is always defined to prevent potential runtime errors. Consider adding a default value or a check to handle cases where `filter.query` might be undefined.